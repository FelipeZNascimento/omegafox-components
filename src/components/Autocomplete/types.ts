export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
