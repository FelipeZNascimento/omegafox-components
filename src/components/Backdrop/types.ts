export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
