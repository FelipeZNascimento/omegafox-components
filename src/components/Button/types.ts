export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
