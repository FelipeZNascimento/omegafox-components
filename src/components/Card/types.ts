export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
