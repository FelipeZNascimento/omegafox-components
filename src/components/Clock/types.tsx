import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
