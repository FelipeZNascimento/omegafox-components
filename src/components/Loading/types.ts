export interface LoadingProps {
  image: string;
  overlay?: boolean;
  shadowed?: boolean;
  size?: 'small' | 'regular' | 'big';
  style?: 'spin' | 'headbutt';
  text?: string;
}
