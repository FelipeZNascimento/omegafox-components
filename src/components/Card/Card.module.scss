@use "../../styling/colors.scss" as colors;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.cardContainer {
  @include color_utils.shadowed(colors.$black, 0.3);
  @include flex-aligned(center, center);

  border-radius: $border-radius;
  border-bottom-left-radius: 0px;
  cursor: pointer;
  flex-direction: column;
  transition: filter $transition-time, background-color $transition-time, box-shadow $transition-time;
  width: 100%;

  &:hover {
    @include color_utils.shadowed(colors.$black, 0.5);
  }
}

.card {
  @include flex-aligned(center, center);

  background-color: rgba(colors.$grey2, 0.6);
  border-radius: $border-radius;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  color: colors.$grey5;
  flex-direction: column;
  height: 60px;
  transition: filter $transition-time, background-color $transition-time;
  width: 100%;

  .title {
    @include marged(all, none);
    @include padded(horizontal, s);
    @include aligned(center);

    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    display: -webkit-box;
    flex-direction: column;
    font-weight: bold;
    overflow: hidden;
    text-overflow: ellipsis;

    &Mobile {
      @include font-sized(l);
    }

    &Desktop {
      @include font-sized(l3);
    }
  }

  .subtitle {
    @include font-sized(s);
    @include marged(vertical, s2);
  }

  &Selected {
    background-color: rgba(colors.$grey3, 0.6);
  }
}

.status {
  @include flex-aligned(center, center);
  @include padded(horizontal, s2);
  @include padded(vertical, s2);

  background-color: rgba(colors.$grey1, 0.4);
  border-bottom-right-radius: $border-radius;
  box-sizing: border-box;
  flex: 0;
  min-height: 60px;
  width: 100%;
}
