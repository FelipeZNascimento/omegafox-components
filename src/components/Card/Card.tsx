import React from 'react';
import { isMobile } from 'react-device-detect';

import classNames from 'classnames';
import { ICardProps } from './types';
import styles from './Card.module.scss';

export const Card = ({
  isForceMobile = false,
  isSelected,
  title,
  subtitle = null,
  onClick = null,
  renderingStatusFunction = null
}: ICardProps) => {
  const cardContainerClass = classNames(styles.cardContainer, {
    [styles.cardContainerSelected]: isSelected
  });

  const cardClass = classNames(styles.card, {
    [styles.cardSelected]: isSelected
  });

  const titleClass = classNames(styles.title, {
    [styles.titleMobile]: isMobile || isForceMobile,
    [styles.titleDesktop]: !isMobile && !isForceMobile
  });

  const handleOnClick = () => {
    if (!onClick) {
      return;
    }

    onClick();
  };

  return (
    <div className={cardContainerClass} onClick={handleOnClick}>
      <div className={cardClass}>
        <p className={titleClass}>{title}</p>
        {subtitle && <span className={styles.subtitle}>{subtitle}</span>}
      </div>
      <div className={styles.status}>
        {renderingStatusFunction ? renderingStatusFunction() : null}
      </div>
    </div>
  );
};
