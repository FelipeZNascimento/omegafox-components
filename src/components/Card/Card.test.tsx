import React from 'react';
import { render } from '@testing-library/react';
import { Card } from './Card';
import { ICardProps } from './types';

describe('<Card />', () => {
  const renderComponent = ({
    isSelected,
    title,
    subtitle = null,
    onClick = null
  }: ICardProps) =>
    render(
      <Card
        isSelected={isSelected}
        title={title}
        subtitle={subtitle}
        onClick={onClick}
      />
    );

  it('should render', () => {
    const isSelected = true;
    const title = 'Title';
    const subtitle = 'Subtitle';
    const { container } = renderComponent({
      isSelected: isSelected,
      title: title,
      subtitle: subtitle
    });

    expect(container).toMatchSnapshot();
  });
});
