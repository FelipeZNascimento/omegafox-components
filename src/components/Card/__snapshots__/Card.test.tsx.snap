// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render 1`] = `
<div>
  <div
    class="cardContainer cardContainerSelected"
  >
    <div
      class="card cardSelected"
    >
      <p
        class="title titleDesktop"
      >
        Title
      </p>
      <span
        class="subtitle"
      >
        Subtitle
      </span>
    </div>
    <div
      class="status"
    />
  </div>
</div>
`;
